src/file.rs

```diff
@@ -53,7 +55,7 @@ impl TryFrom<File> for Header {
 
         file.rewind()?;
 
-        let mut handle = file.take(Header::len());
+        let mut handle = file.take(Header::len() as u64);
```

Comment:
I haven't seen take() used on a file before. I assume it acts like it does on a vector,.
How does it identify the Header in the file, per say.

Reply:
It does it by loading just the bytes requested into memory, without loading the entire file into memory.